### ISSUE TITLE
Adding a column and pick the source name file (csv)

### ISSUE THREAD

**Issue by tigerfoot** [dimitri/pgloader#827]
```
    pgloader --version
pgloader version "3.4.1"
compiled with SBCL 1.4.0-1.el7
```

 ~~~
LOAD CSV
   FROM ALL FILENAMES
        MATCHING ~/export_testme_.*.csv/
        IN DIRECTORY '/repository/data/' WITH ENCODING LATIN-1
   INTO postgresql:///testme?import.csv
   -- truncate during test period
   WITH truncate,
        skip header = 1,
        fields not enclosed,
        fields escaped by backslash-quote,
        fields terminated by '\t'

   SET work_mem to '128 MB', maintenance_work_mem to '64 MB';
~~~

```csv
date_import\tuid\tprice\ttext
22/06/2018\taaavvv\t\100.00\tpromotion
```

```sql
create table import (
 source_name text,
 date_import text,
 uid text,
 price text,
 promotion text
);
```

In import table column name should contain at least import.csv for a csv source named import.csv.20180906

I have a table with a quite big number of colums (120) and in one of the column is not existing into the csv so the idea would be to fill it with some data (I've read the constant column example).
The expected data should be the name of the csv actually parsed + YYYYMMDD date format

I wasn't able to find what can I use as value for the filename (as you certainly pick it, kinda dynamic with the pattern)

I've to insert 4 files per day, so I could eventually define each time the name of the file to load, and make the constant data hardcoded. But from what I've seen in example that would force me to declare all the columns in source and destination ?

Is there a way to pick dynamically the source read (I can replace pattern by fixed named), if yes how ?

Is there a way to simply add a column or specify it's content without declaring all fields ?

**Comment by dimitri** [comment]
Hi @tigerfoot,

  - pgloader currently lacks support for injecting the current filename being loaded as a variable, though it would be a fair feature request to make it available when loading from a file,
  - pgloader supports loading fields from the CSV files in any order in the PostgreSQL table columns, and more than than, supports loaded NF fields in NC columns ; as a result when NF and NC are not the same, you must list the whole set of fields and columns to specify how to match them together.

For the second point, another way to say it is: where is the missing field?

My advice is that you generate the load command from a script with the very specific details of your own implementation / environment / use case. We can also work together and make the filename available to the column expressions through sponsoring of the feature.

**Comment by tigerfoot** [comment]
Ok just a report of progress : First I've opened a bug against pgloader packaged at yum.postgresql.org to be updated to latest version. [pgdg](https://redmine.postgresql.org/issues/3640)
pgloader 3.5.2 has also been updated for openSUSE and will be soon an official package in Tumbleweed (the rolling release) \o/.
Once I will be able to build the documentation with sphinx properly, I guess I will PR here changes for the spec file.

With 3.5.2 and mustache syntax support it was kind easy to script the source name I needed and use the constant column possibility to fill it inside the database. 

Not perfect but working.
